text/0000-pi-types.md

```diff
@@ -0,0 +1,644 @@
+- Feature Name: pi-types
+- Start Date: 2016-06-22
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+We propose a simple, yet sufficiently expressive, addition of dependent-types
+(also known as, Π-types and value-types).
+
+Type checking will not require SMT-solvers or other forms of theorem provers.
+
+## Generic value parameters
+
+A `const` type parameter acts like a generic parameter, containing a constant
+expression. Declaring a generic parameter `const a: usize`, declares a constant
+variable `a` of type `usize`.
+
+One can create implementations, structs, enums, and traits, abstracting over
+this generic value parameter.
```

Comment:
If I'm not missing anything there are right now examples for impls, structs, and functions in the RFC but no examples for enums and traits. It might be worth it to add some and to link the keywords here to the appropiate lines of the RFC so that the reader can jump to those if desired.